client/app/bundles/Restaurants/components/HomePage/Restaurants.jsx

```diff
@@ -0,0 +1,140 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import Header from '../Header/Header';
+import Restaurant from '../RestaurantItem/Restaurant';
+
+//Filters
+import Stars from '../Filters/Stars'
+import CheckBox from '../Filters/Checkbox'
+import RangeBar from '../Filters/RangeBar'
+
+const cuisineTypes = [
+    'What\'s your type?', 'Asian', 'Cafe', 'Italian', 'Hamburgers', 'Salads'
+];
+
+class Restaurants extends React.Component {
+
+    static propTypes = {
+        restaurants: React.PropTypes.array.isRequired
+    };
+
+    constructor() {
+        super();
+        this.state = {filterTenbis: false, filterKosher: false, max_time: 120}
+    }
+
+    onStarClick(nextValue, prevValue, name) {
+        this.setState({filterStars: nextValue});
+    }
+
+    onCuisineSelected(e) {
+        this.setState({filterCuisine: e.target.value})
+    }
+
+    onToggleCheckbox(e) {
+        if (e.target.name === 'isTenbis') {
+            this.setState({filterTenbis: e.target.checked})
+        } else if (e.target.name === 'isKosher') {
+            this.setState({filterKosher: e.target.checked})
+        }
+    }
+
+    onMaxTimeUpdate(e) {
+        this.setState({
+            max_time: e.target.value
+        })
+    }
+
+    render() {
+
+        //todo change
+        let mapPlaceholderUrl = "http://dailygenius.com/wp-content/uploads/2016/04/google-maps-new-interface1.jpg";
+
+        let rests = this.props.restaurants;
+        //filter by rating
+        if (this.state.filterStars) {
```

Comment:
I dont think the filtering logic should be in Render
Render should only render...